08/05/24 category sidebar fix

daniel-sussman opened the pull request.

daniel-sussman committed:
fixed PinpointHQ incorrect ats_job_id

daniel-sussman committed:
get_application_fields from api data for Recruitee

daniel-sussman committed:
modified job_application partial to display Recruitee fields

daniel-sussman committed:
Merge branch 'master' of github.com:Ches-ctrl/Cheddar into 07/05/24_recruitee_apply_with_cheddar

daniel-sussman committed:
include relevant ATS module in form_filler

daniel-sussman committed:
added field_label to ApplicationResponse, added labels to gh and recruitee fields

daniel-sussman committed:
improved category_sidebar load times with sql queries

daniel-sussman committed:
major revision to category_sidebar to improve load speed

daniel-sussman committed:
Bullet has been defeated.

daniel-sussman committed:
cleaned up front-end

daniel-sussman committed:
Merge branch 'master' into 08/05/24_category_sidebar_fix

daniel-sussman committed:
incorporated PR#151 category_sidebar_remove_company

Ches-ctrl commented on app/models/concerns/ats/greenhouse/application_fields.rb:
I'm wondering why we still get location_click when scraping the form fields. One to perhaps discuss with Ilya as well - ideally we have a perfect scrape.

daniel-sussman replied:
Think this is necessary for Capybara to operate the form correctly, but yes Ilya knows more about it.

Ches-ctrl commented on app/models/concerns/ats/greenhouse/application_fields.rb:
Also nice that we're pulling directly from the data_url

Ches-ctrl commented on app/models/concerns/ats/recruitee/application_fields.rb:
Can we go through this later as to how it works

Ches-ctrl commented on app/models/job.rb:
What are the deletes here doing?

daniel-sussman replied:
First delete removes :company from the param filters, since it's no longer possible to filter by company.
Second delete because there's a built-in Ruby method (#except) that does the same thing as previously.

Ches-ctrl commented on app/controllers/job_applications_controller.rb:
I see you merged in my other PR - how did you do that? And does that mean I can now delete that branch?

daniel-sussman replied:
Manually! But yes, you can delete that branch now.

Ches-ctrl commented on app/services/category_sidebar.rb:
Very interested in how you managed this fix

Ches-ctrl commented on app/services/form_filler.rb:
Helpful comment thanks

Ches-ctrl commented on app/services/form_filler.rb:
Like the idea of caching in the future

Ches-ctrl commented on app/views/job_applications/_job_application_form.html.erb:
In general my understanding is to limit the use of html_safe given its dangerous - think we can remove this here as well

daniel-sussman replied:
Yes, it's potentially dangerous -- but the field_label is data from the ATS, not from users, so unlikely to be malicious. html_safe is necessary in order to display the label with its original html tags. Otherwise you just get a plaintext string full of tags. I think we can use #sanitize as well to avoid risk, but not sure quite what that method does.

Ches-ctrl commented on app/views/job_applications/_job_application_form.html.erb:
How come we now have resume or resume/cv?

daniel-sussman replied:
Thanks for flagging this, I'll correct it.

Ches-ctrl reviewed (changes requested):
One minor change on html_safe and some clarifications we can go through later. THIS IS AWESOME!

daniel-sussman committed:
changed instances of 'resume/cv' in field_name to 'resume'

daniel-sussman committed:
added #sanitize before #html_safe on job_application_form

daniel-sussman commented:
Fixed the 'resume/cv' ambiguity. I added #sanitize before #html_safe in the application_form partial, which I think is the best solution unless we prefer to eliminate styling of application questions.

In terms of _when_ to sanitize the HTML string, would it be better to do that data is collected from the API? My thinking is that there are multiple APIs and therefore multiple sources we'd need to ensure are sanitized, which seems dodgier (and less DRY) than sanitizing in the view.

daniel-sussman committed:
fixed syntax

daniel-sussman committed:
can successfully submit a job application with Recruitee

daniel-sussman committed:
bugfix to prevent ApplyJob from overwriting user changes to core fields

daniel-sussman committed:
removed unsafe URI.open()

daniel-sussman committed:
fixes to GetFormFields to ensure it grabs correct labels for all fields

daniel-sussman committed:
added test for company creation with bad ats_identifiers

daniel-sussman committed:
Delete spec/fixtures/cassettes/ApplicantTrackingSystem/with_the_currently_written_modules/doesn_t_throw_an_error_when_unsuccessful_in_creating_a_company_with_any_ATS.yml

daniel-sussman committed:
Delete spec/fixtures/cassettes/ApplicantTrackingSystem/with_the_currently_written_modules/neither_throws_an_error_nor_persists_non-existent_companies_with_any_ATS.yml

Ches-ctrl reviewed (approved):
Nice!
